spec/requests/organisation_relationships_controller_spec.rb

```diff
@@ -296,6 +296,94 @@
         end
       end
 
+      context "when directly accessing the page to add a stock owner" do
+        let(:request) { get "/organisations/#{organisation.id}/stock-owners/add", headers: }
+
+        it "returns 401" do
+          request
+          expect(response).to have_http_status(:unauthorized)
+        end
+      end
+
+      context "when directly adding a stock owner" do
+        let!(:stock_owner) { FactoryBot.create(:organisation) }
+        let(:params) do
+          {
+            "organisation_relationship": {
+              "parent_organisation_id": stock_owner.id,
+            },
+          }
+        end
+        let(:request) { post "/organisations/#{organisation.id}/stock-owners", headers:, params: }
+
+        it "returns 401" do
+          request
+          expect(response).to have_http_status(:unauthorized)
+        end
+
+        it "does not create a new organisation relationship" do
+          expect { request }.not_to change(OrganisationRelationship, :count)
+        end
+      end
+
+      context "when directly removing a stock owner" do
+        let(:stock_owner) { FactoryBot.create(:organisation) }
+        let(:request) { get "/organisations/#{organisation.id}/stock-owners/remove?target_organisation_id=#{stock_owner.id}", headers: }
+
+        before do
+          FactoryBot.create(:organisation_relationship, parent_organisation: stock_owner, child_organisation: organisation)
+        end
+
+        it "returns 401" do
+          request
+          expect(response).to have_http_status(:unauthorized)
+        end
+      end
+
+      context "when directly accessing the page to add a managing agent" do
+        let(:request) { get "/organisations/#{organisation.id}/managing-agents/add", headers: }
+
+        it "returns 401" do
+          request
+          expect(response).to have_http_status(:unauthorized)
+        end
+      end
+
+      context "when directly adding a managing agent" do
+        let!(:managing_agent) { FactoryBot.create(:organisation) }
+        let(:params) do
+          {
+            "organisation_relationship": {
+              "child_organisation_id": managing_agent.id,
+            },
+          }
+        end
+        let(:request) { post "/organisations/#{organisation.id}/managing-agents", headers:, params: }
+
+        it "returns 401" do
+          request
+          expect(response).to have_http_status(:unauthorized)
+        end
+
+        it "does not create a new organisation relationship" do
+          expect { request }.not_to change(OrganisationRelationship, :count)
+        end
+      end
+
+      context "when directly removing a managing agent" do
+        let(:managing_agent) { FactoryBot.create(:organisation) }
+        let(:request) { get "/organisations/#{organisation.id}/managing-agents/remove?target_organisation_id=#{managing_agent.id}", headers: }
```

Comment:
from what I've read one of the benefits of using request specs is that it allows you to implicitly test your routing. Would it be possible to use a path helper here?

Reply:
I've also found that in many places where I've tried removing the `headers:` part in these tests it works fine as the headers are just set to the default html at the top and never changed

Reply:
Happy to use helpers instead of urls, but it would make sense to keep the tests consistent. Also happy to discuss this as a tech debt if we think that we want to change all of these.

Removed headers, nice spot!